materialize-sql/driver.go

```diff
@@ -200,6 +189,22 @@ func (d *Driver) Apply(ctx context.Context, req *pm.Request_Apply) (*pm.Response
 		return nil, err
 	}
 
+	if endpoint.MetaCheckpoints != nil && !is.HasResource(endpoint.MetaCheckpoints.Path) {
```

Comment:
I moved the concept of a "metadata table" entirely out of the materialize-boilerplate and into materialize-sql. The checkpoints table is now created here if it is needed.